quinn/src/endpoint.rs

```diff
@@ -199,6 +201,12 @@ where
                 break;
             }
         }
+        if endpoint.incoming_dirty && endpoint.incoming_live {
```

Comment:
Any reason why checking `!incoming.is_empty()` wouldn't work?

Reply:
I was thinking we shouldn't wake the consumer repeatedly if they're not reading, but of course consuming the waker takes care of that already. Good catch.